src/resolver/namer/mod.rs

```diff
@@ -0,0 +1,63 @@
+use super::{Result, Error, WeightedAddr};
+use tokio_core::reactor::Handle;
+use tokio_timer::Timer;
+use futures::Stream;
+use std::collections::HashMap;
+use futures::stream;
+use futures::Poll;
+
+pub mod namerd;
+
+use self::namerd::{Addrs, WithClient, Namerd};
+
+pub enum StreamEither<A, B> {
+    A(A),
+    B(B),
+}
+
+impl<A, B> Stream for StreamEither<A, B>
```

Comment:
This is cute - I _bet_ we could easily generalize this to make an `Either` that implements _any_ trait implemented by both `A` and `B`?